net/net-snmp/Portfile

```diff
@@ -40,6 +39,8 @@ depends_lib             port:bzip2 \
 patchfiles-append       configure-frameworks.patch
 patchfiles-append       fix-order-of-L-flags.patch
 patchfiles-append       Makefile.top-libtool-tag.patch
+# https://trac.macports.org/ticket/70165
+patchfiles-append       patch-legacy-compat.diff
```

Comment:
I'd prefer if this was if'ed to old versions of the OS. I understand that the patch is technically harmless on newer versions, but that would reduce cognitive load in reviewing it as someone uninterested in an old OS could simply ignore it completely rather than reading it and figuring out that it doesn't impact a new OS.

Reply:
Let’s @ryandesign comment on this one.

(I do not think that unnecessarily conditional patches is a great idea, and I regularly have to fix something accidentally broken due to such patches not being dropped or rebased when needed, but I do not want to spend time and energy on trying to convince anyone on this issue.)

Reply:
So from my point of view, the most important thing is reducing cognitive load on the reviewers. Things that cannot impact current versions of MacOS don't need as much scrutiny.

Reply:
@pmetzger Giving commit rights to active contributors would certainly help here.

I (and Kirill too, earlier) have requested commit access several times. That was met with total silence.
And it’s been two plus years.

I have no idea who and how takes decisions here, but if it just because _someone_ happens to suspect that I intend to commit straightaway to the master some weird experimental stuff for `ppc`: well, it should be pretty clear that I have zero incentive to be irresponsible or annoying, as at the very least – besides making others upset unnecessarily – it will lead to an immediate loss of commit rights. Why would I want to do that? There is such a thing as _discipline of repeated dealings,_ it works.

Reply:
I don't know how those decisions are made; I'm not involved. That said, it has nothing to do with my current comment. It's always easier when I can say "this only impacts old PPC machines, I can just glance lightly at it." We live in an era where unfortunately people can and do spend years slowly ingratiating themselves into open source projects before performing supply chain attacks.

Reply:
I get your point , but Ryan has long stated he prefers patches that are always applied.

This allows patch failures to show up on all systems.

Reply:
Yes, and I already filed an upstream bug report for this. They already have a similar patch in another file, making it seem likely to me that they will accept a patch for this too and we won't be carrying our patch around for long.